code/game/machinery/doors/firedoor.dm

```diff
@@ -1,6 +1,10 @@
 #define CONSTRUCTION_PANEL_OPEN 1 //Maintenance panel is open, still functioning
 #define CONSTRUCTION_NO_CIRCUIT 2 //Circuit board removed, can safely weld apart
 #define DEFAULT_STEP_TIME 20 /// default time for each step
+#define MINIMUM_TEMPERATURE_TO_BURN_ARMS 500 ///everything above this temperature will start burning unprotected arms
+#define TOOLLESS_OPEN_DURATION_SOLO (20 SECONDS) ///opening a firelock without a tool takes this long if only one person is doing it
+#define TOOLLESS_BURN_DAMAGE_PER_SECOND 5 ///how much burn
+#define TRUE_OPENING_TIME max(TOOLLESS_OPEN_DURATION_SOLO / max(length(people_trying_to_open) * 0.75, 1), 2 SECONDS)
```

Comment:
To be hygienic, make `people_trying_to_open` an argument of the macro that is provided.

That is, it'll make everything that uses this use `TRUE_OPENING_TIME(people_trying_to_open)`